### PR TITLE
Fix some typos, mostly "that that"

### PR DESCRIPTION
**Pull request by apivovarov**
Fix some typos

### CODE REVIEW

**Comment by janeyx99** [comment]
Hey @apivovarov, thanks for the typo fix. I did a search and noticed that there are many "that that"s that could be shortened--could you address those as a part of this PR as well?

**Comment by apivovarov** [comment]
Not all "that that"s should be fixed. Example where it is ok to use "that that":
```c
      // if cur_index is not the same as the index of db_loop, it must
      // be true that that index has been modified to support
      // unswitch.
```

**Comment by janeyx99** [comment]
Yes, but all the ones in allocator.cu seem to be fixable

File: torch/fx/passes/utils/matcher_utils.py [hunk 1]
```diff
@@ -359,7 +359,7 @@ def backtracking(anchor_index, match):
         if before != after:
             logger.info("Filtered out %s matches because they are not fully contained", before - after)
 
-        # filter out the matches that that forms a cycle if the subgraph is fused
+        # filter out the matches that forms a cycle if the subgraph is fused
```
Comment (janeyx99) [inline torch/fx/passes/utils/matcher_utils.py:362]:
```suggestion
        # filter out the matches that form a cycle if the subgraph is fused
```

**Review by janeyx99 — APPROVED** [review]
just one more typo fix

File: torch/fx/passes/utils/matcher_utils.py (hunk 1 above)
Comment (apivovarov) [reply torch/fx/passes/utils/matcher_utils.py]: fixed

**Comment by apivovarov** [comment]
@janeyx99 Jane, do we need anything else before it can be merged?

**Pull request closed without merging**